### PR TITLE
core reference refactor

### PR DESCRIPTION
**Pull request by ashleyschuett**
@nicktate @normanjoyner

### CODE REVIEW

**Comment by ashleyschuett** [comment]
cluster 'core refactor' under 'containership-integration' is using this PR.

**Comment by nicktate** [comment]
this is not necessary, but if I were you I would have just done `const core = req.core` at the top of every handler.

**Comment by ashleyschuett** [comment]
@nicktate updated to use `const core = req.core`

File: lib/routes.js [hunk 1]
```diff
@@ -3,68 +3,112 @@
...
-    server.put('/:api_version/variables/:variable', middleware.get_handler('variable', 'update'), middleware.handle_response);
+    server.put('/:api_version/variables/:variable',
+        middleware.get_handler('variable', 'update')
+    );
 
     // api delete variable
-    server.delete('/:api_version/variables/:variable', middleware.get_handler('variable', 'delete'), middleware.handle_response);
+    server.delete('/:api_version/variables/:variable',
+        middleware.get_handler('variable', 'delete')
+    );
```
Comment (nicktate) [inline lib/routes.js]:
```
server.delete('/:api_version/variables' +
  '/variables/:variable', middlware.get_handler('variable', 'delete'));
```

File: lib/middleware.js [hunk 2]
```diff
@@ -2,122 +2,114 @@
...
-        const methods = {
-            // ensure auth
-            authed(req, res, next) {
+const handlers = require('../handlers');
+
+module.exports = (core) => {
+
+    return {
+        // ensure auth
+        authed(req, res, next) {
```
Comment (normanjoyner) [inline lib/middleware.js]: can be removed

File: lib/middleware.js [hunk 3]
```diff
@@ -2,122 +2,114 @@
...
+            return next();
```
Comment (nicktate) [inline lib/middleware.js]: 💣 this

File: lib/server.js [hunk 4]
```diff
@@ -33,11 +33,15 @@ Server.prototype.start = function() {
     this.server.use(this.middleware.allow_cors);
     this.server.use(this.middleware.redirect_to_controlling_leader);
     this.server.use(this.middleware.authed);
```
Comment (nicktate) [inline lib/server.js]: 💣 this

File: handlers/v1/cluster.js [hunk 5]
```diff
@@ -1,46 +1,75 @@
...
+        });
+    },
+
+    // get hosts, applications, and other cluster stuff
+    state: function(req, res, next) {
+        async.parallel({
+            applications: (fn) => {
+                applications.get(req, res, () => {
+                    if(res.stash.code !== 200) {
+                        return fn('Error getting applications.');
```
Comment (normanjoyner) [inline handlers/v1/cluster.js]: can we use `new Error()` here

File: handlers/v1/cluster.js [hunk 6]
```diff
@@ -1,46 +1,75 @@
...
+                    }
+
+                    return fn(null, res.stash.body);
+                });
+            },
+            hosts: (fn) => {
+                hosts.get(req, res, () => {
+                    if(res.stash.code !== 200) {
+                        return fn('Error getting hosts.');
```
Comment (normanjoyner) [inline handlers/v1/cluster.js]: can we use `new Error()` here

Comment (normanjoyner) [inline handlers/v1/cluster.js]: attach the status code to err

File: handlers/v1/cluster.js [hunk 7]
```diff
@@ -1,46 +1,75 @@
...
+                    }
+
+                    return fn(null, res.stash.body);
+                });
+            }
+        }, (err, results) => {
+            if(err) {
+                res.stash.code = 500;
```
Comment (normanjoyner) [inline handlers/v1/cluster.js]: check err.code to set status code returned

**Pull request merged**